iron-list.html

```diff
@@ -1239,23 +1241,22 @@
           var modulus = vidx % this._itemsPerRow;
           var x = Math.floor((modulus * this._itemWidth) + rowOffset);
           this.translate3d(x + 'px', y + 'px', 0, this._physicalItems[pidx]);
+
+          if (this._shouldRenderNextRow(pidx)) {
+            y += this._rowHeight;
+          }
         } else {
           this.translate3d(0, y + 'px', 0, this._physicalItems[pidx]);
-        }
-
-        if (this._shouldRenderNextRow(this.grid ? vidx : pidx)) {

```

Comment:
Great question!

When iterating through the list from the beginning the value of `pidx` accurately represents the row breaks in grid mode. However based on the size of the screen the row breaks (as computed by `pidx`) can occur at the wrong points in the actual list itself. This results in occasionally stacked items. By computing based on `vidx` we ensure that we are always attempting to move to the next row at the proper "row breaks."